web/src/compose_notifications.ts

```diff
@@ -245,6 +248,71 @@ export function notify_messages_outside_current_search(messages: Message[]): voi
     }
 }
 
+export function maybe_show_one_time_messages_fading_banner(): void {
+    const is_faded = $(".recipient_row").hasClass("message-fade");
+
+    if (is_faded) {
+        let recipient;
+        let narrow_url;
+        if (compose_state.get_message_type() === "stream") {
+            const stream_name = compose_state.stream_name();
+            const topic_name = compose_state.topic();
+            const stream_id = compose_state.stream_id();
+            assert(topic_name !== undefined);
+            assert(stream_id !== undefined);
+
+            recipient = `#${stream_name} > ${topic_name}`;
+
+            narrow_url = hash_util.by_stream_topic_url(stream_id, topic_name);
+        } else {
+            const emails_string = compose_state.private_message_recipient();
+            const user_ids_string = people.emails_strings_to_user_ids_string(emails_string);
+            assert(user_ids_string !== undefined);
+            const user_ids = people.user_ids_string_to_ids_array(user_ids_string);
+            const names = people.get_display_full_names(user_ids).sort();
+
+            recipient = names.join(", ");
```

Comment:
You can use `util.format_array_as_list` here to get a better result (it'll use "and" to make a nicer conjunction).

Reply:
![image](https://github.com/zulip/zulip/assets/56781761/65fea67e-2d6f-49bc-93b5-2f7ebfddc304)